enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -418,19 +418,24 @@ func (l *FileCacheLoader) GetSnapshot(ctx context.Context, keys *fcpb.SnapshotKe
 func (l *FileCacheLoader) getSnapshot(ctx context.Context, key *fcpb.SnapshotKey, remoteEnabled bool) (*fcpb.SnapshotManifest, error) {
 	ctx, span := tracing.StartSpan(ctx)
 	defer span.End()
-	if *snaputil.EnableRemoteSnapshotSharing && remoteEnabled {
-		manifest, err := l.fetchRemoteManifest(ctx, key)
-		if err != nil {
-			return nil, status.WrapError(err, "fetch remote manifest")
+
+	// Always prioritize the local manifest if it exists. It may point to a more
+	// updated snapshot than the remote manifest, which is updated less frequently.
```

Comment:
This extra context would be helpful to understand why the local snapshot is always more up to date, I think:

```suggestion
	// updated snapshot than the remote manifest, which is updated less frequently.
	// Note that the master snapshot is never cached locally.
```